frappe/model/db_query.py

```diff
@@ -423,20 +423,25 @@ def get_share_condition(self):
 	def add_user_permissions(self, user_permissions, user_permission_doctypes=None):
 		user_permission_doctypes = frappe.permissions.get_user_permission_doctypes(user_permission_doctypes, user_permissions)
 		meta = frappe.get_meta(self.doctype)
-
 		for doctypes in user_permission_doctypes:
 			match_filters = {}
 			match_conditions = []
 			# check in links
 			for df in meta.get_fields_to_check_permissions(doctypes):
 				user_permission_values = user_permissions.get(df.options, [])
 
-				condition = 'ifnull(`tab{doctype}`.`{fieldname}`, "")=""'.format(doctype=self.doctype, fieldname=df.fieldname)
+				cond = 'ifnull(`tab{doctype}`.`{fieldname}`, "")=""'.format(doctype=self.doctype, fieldname=df.fieldname)
 				if user_permission_values:
-					condition += """ or `tab{doctype}`.`{fieldname}` in ({values})""".format(
+					if not cint(frappe.db.get_single_value("System Settings", "apply_strict_user_permissions")):
```

Comment:
why query a near static value inside a loop? this should be ideally cached, atleast locally

Commonify this: https://github.com/frappe/frappe/blob/develop/frappe/__init__.py#L1367 as `frappe.get_system_setting`

So that it can be used for any system value